pkg/ctl/create/nodegroup.go

```diff
@@ -0,0 +1,137 @@
+package create
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/kubicorn/kubicorn/pkg/logger"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/eks/api"
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	"github.com/pkg/errors"
+)
+
+func createNodeGroupCmd() *cobra.Command {
+	p := &api.ProviderConfig{}
+	cfg := api.NewClusterConfig()
+	ng := cfg.NewNodeGroup()
+
+	cmd := &cobra.Command{
+		Use:   "nodegroup",
+		Short: "Create a nodegroup",
+		Run: func(_ *cobra.Command, args []string) {
+			if err := doAddNodeGroup(p, cfg, ng, cmdutils.GetNameArg(args)); err != nil {
+				logger.Critical("%s\n", err.Error())
+				os.Exit(1)
+			}
+		},
+	}
+
+	fs := cmd.Flags()
+
+	addCommonCreateFlags(fs, p, cfg, ng)
+	fs.StringVarP(&cfg.Metadata.Name, "cluster", "n", "", "Name of the EKS cluster to add the nodegroup to")
+
+	return cmd
+}
+
+func doAddNodeGroup(p *api.ProviderConfig, cfg *api.ClusterConfig, ng *api.NodeGroup, name string) error {
+	ctl := eks.New(p, cfg)
+
+	if cfg.Metadata.Region != api.EKSRegionUSWest2 && cfg.Metadata.Region != api.EKSRegionUSEast1 && cfg.Metadata.Region != api.EKSRegionEUWest1 {
+		return fmt.Errorf("%s is not supported only %s, %s and %s are supported", cfg.Metadata.Region, api.EKSRegionUSWest2, api.EKSRegionUSEast1, api.EKSRegionEUWest1)
+	}
+
+	if err := ctl.CheckAuth(); err != nil {
+		return err
+	}
+
+	if cfg.Metadata.Name == "" {
+		return errors.New("--cluster must be specified. run `eksctl get cluster` to show existing clusters")
+	}
+
+	if ng.SSHPublicKeyPath == "" {
+		return fmt.Errorf("--ssh-public-key must be non-empty string")
+	}
+
+	//TODO:  do we need to do the AZ stuff from create????
```

Comment:
I can't remember why i added that comment.

Reply:
Perhaps you wanted to confirm the new functionally supports `--node-zones` to select which zone(s) to use for the nodegroup?

AFAICS, it just works. `--node-zones` populates `AvailabilityZones` in the nodegroup spec, which is then handled by the cfn builder to either select subnets that corresponds to the zones, or import: https://github.com/weaveworks/eksctl/blob/537f2c413330269cf044ee04f8a1a42058231235/pkg/cfn/builder/nodegroup.go#L141-L162

Reply:
Yes thats right. I have a feeling that @errordeveloper commented back then that for a new nodegroup it would be good to be able to use specific zones (and to overide the default behavior).